faucet/faucet_dot1x.py

```diff
@@ -144,124 +141,87 @@ def set_mac_str(self, valve, valve_index, port_num):
         self.mac_to_port[mac_str] = (valve, port)
         return mac_str
 
-    def port_up(self, valve, dot1x_port):
-        """Setup the dot1x forward port acls.
+    def nfv_sw_port_up(self, dp_id, dot1x_ports, nfv_sw_port, acl_manager):
+        """Setup the dot1x forward port acls when the nfv_sw_port comes up.
         Args:
-            dot1x_port:
-            valve:
+            dp_id (int):
+            dot1x_ports (Iterable of Port objects):
+            nfv_sw_port (Port):
+            acl_manager (ValveAclManager):
 
         Returns:
             list of flowmods
         """
-        self.dot1x_speaker.port_down(get_mac_str(self.dp_id_to_valve_index[valve.dp.dp_id],
-                                                 dot1x_port.number))
-        nfv_sw_port = valve.dp.dot1x['nfv_sw_port']
+        self.dot1x_speaker.port_down(
+            get_mac_str(self.dp_id_to_valve_index[dp_id], nfv_sw_port.number))
 
-        if dot1x_port.number == nfv_sw_port:
-            ret = []
-            for port in valve.dp.dot1x_ports():
-                ret.extend(
-                    self.create_flow_pair(port, nfv_sw_port, valve))
-            return ret
+        ret = []
+        for port in dot1x_ports:
+            ret.extend(self.create_flow_pair(
+                dp_id, port, nfv_sw_port, acl_manager))
+        return ret
 
-        return self.create_flow_pair(dot1x_port, nfv_sw_port, valve)
+    def port_up(self, dp_id, dot1x_port, nfv_sw_port, acl_manager):
+        """Setup the dot1x forward port acls.
+        Args:
+            dp_id (int):
+            dot1x_port (Port):
+            nfv_sw_port (Port):
+            acl_manager (ValveAclManager):
 
-    def create_flow_pair(self, dot1x_port, nfv_sw_port, valve):
-        """Creates the pair of flows that redirects the eapol packets to/from the supplicant and
-        nfv port
+        Returns:
+            list of flowmods
+        """
+        self.dot1x_speaker.port_down(
```

Comment:
This was an error on my part and should have been port_up, think @MichaelWasher has fixed it in pr #2697
But seeing how this PR is not adding functionality, I'm guessing someone will refactor later.